### PR TITLE
Add support for Custom Fields.

### PR DESCRIPTION
**Pull request by launeh**
# Description
This PR adds support for custom fields. The highlights are: 

- Added a new CustomFieldsHandler
- Updated the existing checkbox handler to not process custom 'checkbox fields'
- Added a CustomFieldsHandlerTest
- Modified the OrderTest

There is an existing feature switch that controls whether merchants have access to this feature. 

Note: There is a bit of code subscribing to newsletters that's duplicated across the old checkboxes handlers and the customfield handlers. This will be refactored in a separate PR. 

Fixes: https://boltpay.atlassian.net/browse/M2P-472

#changelog Laune/m2 p 472 custom fields

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Helper/Order.php [hunk 1]
```diff
@@ -337,15 +343,16 @@ public function __construct(
...
+        CustomFieldsHandler $customFieldsHandler,    
         CustomerCreditCardFactory $customerCreditCardFactory,
         CustomerCreditCardCollectionFactory $customerCreditCardCollectionFactory,
         CreditmemoFactory $creditmemoFactory,
         CreditmemoManagementInterface $creditmemoManagement,
         EventsForThirdPartyModules $eventsForThirdPartyModules,
         OrderManagementInterface $orderManagement = null,
         OrderIncrementIdChecker $orderIncrementIdChecker = null,
-        Create $adminOrderCreateModel = null
+        Create $adminOrderCreateModel = null
```
Comment (vitaliyreznikov) [inline Helper/Order.php:355]: nit: remove spaces

**Review by vitaliyreznikov — DISMISSED** [review]
LGTM Please update PR title

File: Helper/Order.php (hunk 1 above)
Comment (launeh) [reply Helper/Order.php]: Removed.

Comment (launeh) [reply Helper/Order.php]: And updated PR title.

**Review by vitaliyreznikov — APPROVED** [review]

**Review by danac-gs — APPROVED** [review]

**Pull request merged**